src/Planner/Planner.cpp

```diff
@@ -528,6 +517,17 @@ void Planner::buildQueryPlanIfNeeded()
 
             query_plan.addStep(std::move(totals_having_step));
         }
+
+        if (query_node.isGroupByWithRollup())
```

Comment:
It means that `HAVING` will not be executed for rows produced from `CUBE` and `ROLLUP`. Also, you need to add such a test.

Reply:
HAVING is not supported when there is `WITH TOTALS` and `CUBE` or `ROLLUP`.